example/misc/infinite_set.cpp

```diff
@@ -49,6 +50,11 @@ constexpr auto doubleton(X x, Y y) {
 
 namespace boost { namespace hana {
     template <>
+    struct SetTheoretic<infinite_set_tag> {
+        static constexpr bool value = true;
+    };
```

Comment:
I was thinking it the other way around. If you want to use any of the set theoretic functions (at least one of 4 available at the moment), then it should have a concept `SetTheoretic`. Container doesn't need to implement all 4 operations IMO.
Essentially it means that if you want to use any of these 4 functions, container must be `SetTheoretic`.